### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by regislebrun**
This PR proposes various fixes/improvements:
  * StandardDistributionPolynomialFactory is now able to identify the Binomial case
  * Fixed a segfault in HyperbolicAnisotropicEnumerateFunction::inverse() when the evaluation cache is empty. It fixes issue #1129.
  * Replaced remaining uses of double indexing of Sample in a few classes
  * Added an illustration in solver documentation (Brent, Bisection, Secant), it fixes issue #1080 
  * Made CopulaImplementation derives from DistributionImplementation. It fixes issue #1139.
  * Fixed a circular call in RandomMixture. It fixes issue #1129.
  * Improved the simplification mechanism in TruncatedDistribution, fixing an infinite waiting time in RandomMixture
  * Fixed CalibrationResult by providing additional missing information. It fixes issue #1119.
  * Fixed GEV parameter gradient (see issue #1058)

### CODE REVIEW

File: lib/src/Uncertainty/Bayesian/GaussianNonLinearCalibration.cxx [hunk 1]
```diff
@@ -81,9 +82,10 @@ GaussianNonLinearCalibration::GaussianNonLinearCalibration(const Function & mode
     algorithm_ = OptimizationAlgorithm::Build(leastSquaresNames[0]);
   else
     algorithm_ = MultiStart(TNC(), LowDiscrepancyExperiment(SobolSequence(), Normal(candidate, CovarianceMatrix(candidate.getDimension())), ResourceMap::GetAsUnsignedInteger("NonLinearLeastSquaresCalibration-MultiStartSize")).generate());
+  parameterPrior_.setDescription(model.getParameterDescription());
 }
 
-namespace ThreeDVARFunctions
+namespace GaussianNonLinearFunctions
```
Comment (mbaudin47) [inline lib/src/Uncertainty/Bayesian/GaussianNonLinearCalibration.cxx]: Excellent: I also saw this point, but have not published the PR yet.

File: lib/src/Uncertainty/Algorithm/OrthogonalBasis/StandardDistributionPolynomialFactory.cxx [hunk 2]
```diff
@@ -196,6 +196,12 @@ void StandardDistributionPolynomialFactory::checkSpecificFamily()
     referenceFamily = KrawtchoukFactory(static_cast<UnsignedInteger>(parameter[0]), parameter[1]);
     hasClassMatch = true;
   }
+  if (measureType == "Bernoulli")
```
Comment (mbaudin47) [inline lib/src/Uncertainty/Algorithm/OrthogonalBasis/StandardDistributionPolynomialFactory.cxx]: The "switch" C++ statement would be more appropriate, isn't it ?

Comment (regislebrun) [reply lib/src/Uncertainty/Algorithm/OrthogonalBasis/StandardDistributionPolynomialFactory.cxx]: Doesn't work with complex types, only simple types (int, char & co)

File: lib/test/t_FisherSnedecor_std.cxx [hunk 3]
```diff
@@ -79,7 +79,7 @@ int main(int, char *[])
     Scalar eps = 1e-5;
     Point DDF = distribution.computeDDF( point );
     fullprint << "ddf     =" << DDF << std::endl;
-    fullprint << "ddf (FD)=" << distribution.ContinuousDistribution::computeDDF(point) << std::endl;
+    fullprint << "ddf (FD)=" << distribution.DistributionImplementation::computeDDF(point) << std::endl;
```
Comment (jschueller) [inline lib/test/t_FisherSnedecor_std.cxx]: It doesn't like this is even needed: the test succeeds with `distribution.computeDDF`

File: lib/src/Uncertainty/Distribution/RandomMixture.cxx [hunk 4]
```diff
@@ -2293,32 +2293,17 @@ Scalar RandomMixture::computeProbability(const Interval & interval) const
...
-    for (UnsignedInteger i = 0; i < gridX.getSize(); ++i)
-      if (interval.contains(gridX[i])) cdf += samplePDF(i, 0) * elementaryVolume;
-    return cdf;
-#endif
+    if (isContinuous()) return computeProbabilityContinuous(interval);
+    // Generic implementation for discrete distributions
+    if (isDiscrete())   return computeProbabilityDiscrete(interval);
+    // Generic implementation for general distributions
+    return computeProbabilityGeneral(interval);
   }
```
Comment (jschueller) [inline lib/src/Uncertainty/Distribution/RandomMixture.cxx]: Does it fix #1129 ? if so the changelog must be updated.

**Comment by jschueller** [comment]
Please add tests for the random mixture fix and enumerate function fixes.

I'm not sure about the LinearFunction enhancement, how about my suggestion to add a Null/ZeroHessian class instead ?

The rest is good to go.

File: lib/test/t_FisherSnedecor_std.cxx (hunk 3 above)
Comment (regislebrun) [reply lib/test/t_FisherSnedecor_std.cxx]: It was an attempt to embed a validation of the analytical formula (distribution.computeDDF()) and a finite differences approximation (distribution.DistributionImplementation::computeDDF()). I can remove it if you think it is useless.

File: lib/src/Uncertainty/Distribution/RandomMixture.cxx (hunk 4 above)
Comment (regislebrun) [reply lib/src/Uncertainty/Distribution/RandomMixture.cxx]: Yes it fixes it. I modify the changelog asap.

**Comment by regislebrun** [comment]
Adding a new class looked a little bit overkill to me for the purpose. I wanted to mimic what we do in several places (e.g. Normal with independent components): the trivial case is treated in a specific way without the need of a new class (such as an IndependentMultivariateNormal). But I can add the new class, its tests and its documentation later: we simply drop this commit from the PR.

**Comment by jschueller** [comment]
ok

File: lib/test/t_FisherSnedecor_std.cxx (hunk 3 above)
Comment (jschueller) [reply lib/test/t_FisherSnedecor_std.cxx]: if you can, use sed as it shows up in about 50 tests

**Comment by regislebrun** [comment]
The failure on VS10 is a timeout:
```
99% tests passed, 1 tests failed out of 467
Total Test time (real) = 287.47 sec
The following tests FAILED:
	213 - pyinstallcheck_BoxCoxFactory_glm (Timeout)
Errors while running CTest
```
Anything else wrong?

**Pull request merged**